### PR TITLE
feat: GraphQL application contact

### PR DESCRIPTION
**Pull request by mayaisrani**
recreating #399 

including all field validations

### CODE REVIEW

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:23]: I noticed in the ASPX looks like this field is a dropdown (https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/ApplicationContact.aspx#L233). Does it end up getting converted to a boolean in the database somewhere?

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:31]: State looks to be a dropdown in the ASPX (https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/ApplicationContact.aspx#L337). We should set this as an ENUM.

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:45]: State looks to be a dropdown in the ASPX (https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/ApplicationContact.aspx#L337). We should set this as an ENUM.

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:33]: Zip codes look to have some additional regex validation. Is the legacy ASPX using a single form field for zip and breaking into two separate db fields? If so, we should set up these fields to work with that validation, i.e. zip should also have minLength = 5 and zip4 should have max/min = 4.

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:47]: Zip codes look to have some additional regex validation. Is the legacy ASPX using a single form field for zip and breaking into two separate db fields? If so, we should set up these fields to work with that validation, i.e. zip should also have minLength = 5 and zip4 should have max/min = 4.

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
Added some more comments around validation. We should also write some tests to assert the validations, regex works, etc.

File: oneapp-server/types/ApplicationContact.js
Comment (mayaisrani) [reply oneapp-server/types/ApplicationContact.js:23]: I think this is something we'll want to think through as we build that final call between the frontend and GraphQL. Not sure how much validation/conversion we'll want to do in the GraphQL mutation layer. My intuition is that we do a quick string to boolean conversion in the frontend, so that if/when there are any string changes on the frontend, we won't need to touch the GraphQL/backend

Comment (ronaldheft-usds) [reply oneapp-server/types/ApplicationContact.js:23]: Agreed. Keeping presentation logic out of the API layer is ideal.

I was more thinking about how the database is storing boolean values. It looks like some fields in the database are using a 1 byte varchar stored as `Y/N` and `IS_HOMELESS` appears to be one. I'm wondering if there is a straightforward way we can convert those without having to go field by field. Can we use / write some GraphQL middleware to have the API interface be a Boolean but the underlying data operates on a `Y/N` string?

File: oneapp-server/types/ApplicationContact.js
Comment (ronaldheft-usds) [inline oneapp-server/types/ApplicationContact.js:133]: I'm wondering if there are every situations where a non-NJ address is valid. Like could someone be a NJ resident but living with family in PA and receive the benefits at that PA address? Thinking we should define out all the states as valid in the ENUM.

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
Added some more thoughts around a couple of things I saw in the code.

Getting excited to see the GraphQL API connected to the front-end Mike's worked on. Thanks for all your work on this!

**Review by ronaldheft-usds — CHANGES_REQUESTED** [review]
Core functionality looks great! We just need some tests for the `application.contact` query and `applicationUpdate` mutation and this should be set to merge.

File: oneapp-server/types/Application.test.js
Comment (ronaldheft-usds) [inline oneapp-server/types/Application.test.js:5]: We shouldn't need this dependency in the tests. That was in the user tests to mock out the password reset process, but that piece of code shouldn't be touched in these tests.

File: oneapp-server/types/Application.test.js
Comment (ronaldheft-usds) [inline oneapp-server/types/Application.test.js:8]: Cool! I like this approach to breaking down the `applicationUpdate` tests.

File: oneapp-server/types/Application.test.js
Comment (ronaldheft-usds) [inline oneapp-server/types/Application.test.js:44]: Good call on adding some tests to verify constraints!

**Review by ronaldheft-usds — APPROVED** [review]
This all looks great! I would remove the small piece of unused copy/pasted code in the test before merging, but this is good to merge.

**Pull request merged**
